worker/src/ee/cloudSpendAlerts/handleCloudSpendAlertJob.ts

```diff
@@ -0,0 +1,258 @@
+import { parseDbOrg, Role } from "@langfuse/shared";
+import { prisma } from "@langfuse/shared/src/db";
+import Stripe from "stripe";
+import { env } from "../../env";
+import { logger } from "@langfuse/shared/src/server";
+import { recordIncrement, traceException } from "@langfuse/shared/src/server";
+import { Job } from "bullmq";
+import { backOff } from "exponential-backoff";
+import { sendCloudSpendAlertEmail } from "@langfuse/shared/src/server";
+
+export const handleCloudSpendAlertJob = async (job: Job<{ orgId: string }>) => {
+  const { orgId } = job.data;
+
+  logger.info(`[CLOUD SPEND ALERTS] Processing org ${orgId}`);
+
+  if (!env.STRIPE_SECRET_KEY) {
+    logger.warn("[CLOUD SPEND ALERTS] Stripe secret key not found");
+    throw new Error("Stripe secret key not found");
+  }
+
+  // Fetch organization with spend alerts
+  const orgData = await prisma.organization.findFirst({
+    where: {
+      id: orgId,
+    },
+    include: {
+      cloudSpendAlerts: true,
+    },
+  });
+
+  if (!orgData) {
+    logger.error(`[CLOUD SPEND ALERTS] Organization ${orgId} not found`);
+    return;
+  }
+
+  const org = {
+    ...parseDbOrg(orgData),
+    spendAlerts: orgData.cloudSpendAlerts,
+  };
+
+  // Check if org has spend alerts configured
+  if (org.spendAlerts.length === 0) {
+    logger.info(`[CLOUD SPEND ALERTS] No spend alerts for org ${orgId}`);
+    return;
+  }
+
+  if (org.cloudConfig?.plan === "Hobby") {
+    // handle case where user has downgraded to hobby
+    logger.info(
+      `[CLOUD SPEND ALERTS] Org ${orgId} not entitled to spend alerts (plan: Hobby"})`,
+    );
+    return;
+  }
+
+  // Get Stripe customer ID
+  const stripeCustomerId = org.cloudConfig?.stripe?.customerId;
+  if (!stripeCustomerId) {
+    logger.error(
+      `[CLOUD SPEND ALERTS] Stripe customer id not found for org ${orgId}`,
+    );
+    traceException(
+      `[CLOUD SPEND ALERTS] Stripe customer id not found for org ${orgId}`,
+    );
+    return;
+  }
+  // Get Stripe subscription ID
+  const stripeSubscriptionId = org.cloudConfig?.stripe?.activeSubscriptionId;
+  if (!stripeSubscriptionId) {
+    logger.error(
+      `[CLOUD SPEND ALERTS] Stripe subscription id not found for org ${orgId}`,
+    );
+    traceException(
+      `[CLOUD SPEND ALERTS] Stripe subscription id not found for org ${orgId}`,
+    );
+    return;
+  }
+
+  // Setup stripe client
+  const stripe = new Stripe(env.STRIPE_SECRET_KEY);
```

Comment:
I thought we had a global singleton for this purpose.

Reply:
The singleton exists only in the context of the web container. Should we also create one for the worker?